gulpfile.js/build.js

```diff
@@ -0,0 +1,411 @@
+ *
+ * @return {Stream}
+ */
+function collectStatics(done) {
+  // Used to keep track of unfinished archives
+  const archives = {};
+
+  gulp.src([
+    project.absolute('pages/static/**/*'),
+    project.absolute('examples/static/**/*'),
+  ]).pipe(through.obj(async function(file, encoding, callback) {
+    // Skip potential archive parent directories to have the path writable later
+    if (file.stat.isDirectory() && file.path.endsWith('.zip')) {
+      callback();
+      return;
+    }
+
+    // Check if file could be part of a ZIP and not already is one itself
+    if (file.path.includes('.zip') && !file.path.endsWith('.zip')) {
+      // If the file should be part of a ZIP file pass it over to archiver
+      const relativePath = file.relative.slice(0, file.relative.indexOf('.zip') + 4);
+      const archive = archives[relativePath] || archiver('zip', {
+        'zlib': {'level': 9},
+      });
+
+      // Only append real files, directories will be created automatically
+      const filePath = file.relative.replace(relativePath, '');
+      if (!file.stat.isDirectory() && filePath) {
+        archive.append(file.contents, {'name': filePath});
+      }
+
+      archives[relativePath] = archive;
+      callback();
+      return;
+    }
+
+    // ... and simply copy all other files
+    // eslint-disable-next-line no-invalid-this
+    this.push(file);
+    callback();
+  }))
+      .pipe(gulp.dest(project.paths.STATICS_DEST))
+      .on('end', async () => {
+        signale.await('Writing ZIPs ...');
+
+        const writes = [];
+        for (const [archivePath, contents] of Object.entries(archives)) {
+          contents.finalize();
+
+          const dest = path.join(project.paths.STATICS_DEST, archivePath);
+          const archive = fs.createWriteStream(dest);
+
+          writes.push(new Promise((resolve, reject) => {
+            contents.pipe(archive).on('close', () => {
+              signale.success(`Wrote archive ${archivePath}`);
+              resolve();
+            });
+          }));
+        };
+
+        await Promise.all(writes);
+        signale.await('Finished collecting static files!');
+        done();
+      });
+}
+
+/**
+ * Writes information about the current build to a file to be able to
+ * inspect the current version on /who-am-i
+ *
+ * @return {undefined}
+ */
+function persistBuildInfo(done) {
+  const buildInfo = {
+    'number': travis.build.number || null,
+    'at': new Date(),
+    'by': git.user,
+    'environment': config.environment,
+    'commit': {
+      'sha': git.version,
+      'message': git.message,
+    },
+  };
+
+  fs.writeFile(project.paths.BUILD_INFO, yaml.safeDump(buildInfo), done);
+}
+
+exports.clean = clean;
+exports.importAll = importAll;
+exports.buildPlayground = buildPlayground;
+exports.buildBoilerplate = buildBoilerplate;
+exports.buildFrontend = buildFrontend;
+exports.buildSamples = buildSamples;
+exports.buildPages = buildPages;
+
+exports.setupBuild = setupBuild;
```

Comment:
Can we find a better name for setupBuild?

Reply:
You've got a suggestion? Going with the general > specific naming scheme `buildInit`, `buildBootstrap`, `buildStart` could maybe work. I am not really happy with `setupBuild` either.

Reply:
how about `buildSlow` and `buildFast`? At least that looks like the separator between these two tasks. It also might make sense to split tasks into `internal` and `public` (if that makes sense).

Reply:
Not quite: `setupBuild` runs various smaller tasks in the first place for the sharded instances to not need to run them again.

I went with renaming it to `buildPrepare`.